keps/sig-node/20191118-cgroups-v2.md

```diff
@@ -0,0 +1,242 @@
+- [Non Goals](#non-goals)
+- [User Stories](#user-stories)
+- [Implementation Details](#implementation-details)
+  - [Dependencies on OCI and container runtimes](#implementation-details)
+  - [Current status of dependencies](#current-status-of-dependencies)
+  - [Current cgroups usage and the equivalent in cgroups v2](#current-cgroups-usage-and-the-equivalent-in-cgroups-v2)
+- [Proposal](#proposal)
+  - [Option 1: Convert from cgroups v1 settings to v2](#option-1-convert-from-cgroups-v1-settings-to-v2)
+  - [Option 2: Use cgroups v2 throughout the stack](#option-2-use-cgroups-v2-throughout-the-stack)
+- [Risk and Mitigations](#risk-and-mitigations)
+- [Graduation Criteria](#graduation-criteria)
+- [Testing](#testing)
+
+<!-- /toc -->
+
+## Summary
+
+A proposal to add support for cgroups v2 to kubernetes.
+
+## Motivation
+
+The new kernel cgroups v2 API was declared stable more than two years
+ago. Newer features in the kernel such as PSI depend upon cgroups
+v2. groups v1 will eventually become obsolete in favor of cgroups v2.
+Some distros are already using cgroups v2 by default, and that
+prevents Kubernetes from working as it is required to run with cgroups
+v1.
+
+## Goals
+
+This proposal aims to:
+
+*   Add support for cgroups v2 to the Kubelet
+
+## Non-Goals
+
+*	Expose new cgroup2-only features
+*	Dockershim
+*	Plugins support
+
+## User Stories
+
+*   The Kubelet can run on a host using either cgroups v1 or v2.
+*   Have features parity between cgroup v2 and v1.
+
+## Implementation Details
+
+## Proposal
+
+The proposal is to implement cgroups v2 in two different phases.
+
+The first phase ensures that any configuration file designed for
+cgroups v1 will continue to work on cgroups v2.
+
+The second phase requires changes through the entire stack, including
+the OCI runtime specifications.
+
+At startup the Kubelet detects what hierarchy the system is using.  It
+checks the file system type for `/sys/fs/cgroup` (the equivalent of
+`stat -f --format '%T' /sys/fs/cgroup`).  If the type is `cgroup2fs`
+then the Kubelet will use only cgroups v2 during all its execution.
+
+The current proposal doesn't aim at deprecating cgroup v1, that must
+still be supported through the stack.
+
+Device plugins that require v2 enablement are out of the scope for
+this proposal.
+
+### Dependencies on OCI and container runtimes
+
+In order to support features only available in cgroups v2 but not in
+cgroups v1, the OCI runtime specs must be changed.
+
+New features that are not present in cgroup v1 are out of the scope
+for this proposal.
+
+The dockershim implementation embedded in the Kubelet won't be
+supported on cgroup v2.
+
+### Current status of dependencies
+
+- CRI-O+crun: support cgroups v2
+
+- runc support for cgroups v2 is work in progress [current status](#current-cgroups-usage-and-the-equivalent-in-cgroups-v2)
+
+- containerd: [https://github.com/containerd/containerd/pull/3799](https://github.com/containerd/containerd/pull/3799)
```

Comment:
This has been merged

Reply:
Better to add a link to https://github.com/containerd/containerd/issues/3726

Reply:
thanks, changed to containerd/containerd#3726